### ISSUE TITLE
NoClassDefFoundError com/sun/xml/internal/ws/spi/db/BindingContextFactory

### ISSUE THREAD

**Issue by slm** [mapbox/mapbox-navigation-android#1028]
**Android API: 7.0 - 24**
**Mapbox Navigation SDK version: 0.12.0**

### Steps to trigger behavior

 1. Just see fabric logs :(

### Expected behavior
Don't be crash

### Actual behavior
Be crash.

App use proguard. I think probably cause of proguard settings. 

```
Fatal Exception: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/sun/xml/internal/ws/spi/db/BindingContextFactory;
       at com.mapbox.api.directions.v5.MapboxDirections.errorDidOccur(MapboxDirections.java:192)
       at com.mapbox.api.directions.v5.MapboxDirections.access$000(MapboxDirections.java:60)
       at com.mapbox.api.directions.v5.MapboxDirections$1.onResponse(MapboxDirections.java:144)
       at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:70)
       at android.os.Handler.handleCallback(Handler.java:836)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:203)
       at android.app.ActivityThread.main(ActivityThread.java:6251)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
```

```
Caused by java.lang.ClassNotFoundException: Didn't find class "com.sun.xml.internal.ws.spi.db.BindingContextFactory" on path: DexPathList[[zip file "/data/app/com.taxi724.taximeterv2-2/base.apk"],nativeLibraryDirectories=[/data/app/com.taxi724.taximeterv2-2/lib/arm, /data/app/com.taxi724.taximeterv2-2/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib, /system/vendor/lib]]
       at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
       at com.mapbox.api.directions.v5.MapboxDirections.errorDidOccur(MapboxDirections.java:192)
       at com.mapbox.api.directions.v5.MapboxDirections.access$000(MapboxDirections.java:60)
       at com.mapbox.api.directions.v5.MapboxDirections$1.onResponse(MapboxDirections.java:144)
       at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:70)
       at android.os.Handler.handleCallback(Handler.java:836)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:203)
       at android.app.ActivityThread.main(ActivityThread.java:6251)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
```

**Comment by danesfeder** [comment]
Hey @slm, we will take a look at this and adjust our proguard config if needed.  Thanks for brining this to us! 

@osana this looks like a possible issue with MAS, but I'm not sure - @Guardiola31337 we recently made proguard adjustments in #825 and #838.  Do you think these changes could be the cause of the error above?

**Comment by Guardiola31337** [comment]
@danesfeder 
> this looks like a possible issue with MAS, but I'm not sure

Nah, MAS is pure Java now so it can't include Android Proguard rules.

cc @osana 

---

Hey @slm 👋 thanks for reaching out and report your problem.

As @danesfeder mentioned Proguard issues were solved in #825 and #838 which were included in [`v0.13.0`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v0.13.0) of the SDK. You mentioned in OP that you're still using `0.12.0`. If you upgrade to the latest version of the SDK you shouldn't get that Proguard issue anymore.

Thanks for checking out the SDK!

Closing as answered. Please feel free to reopen if seen again.

**Comment by slm** [comment]
Thanks :)
